### PR TITLE
Allow modify model config at decode time for ASR

### PR DESCRIPTION
**Pull request by iprovalo**
Allow modify model config at decode time for ASR.

https://github.com/k2-fsa/sherpa-onnx/issues/1116

Also, exposes the language which was recognized by whisper to the calling client in the SherpaOnnxOfflineRecognizerResult.

### CODE REVIEW

File: sherpa-onnx/c-api/c-api.h [hunk 1]
```diff
@@ -491,7 +496,9 @@ SHERPA_ONNX_API void DecodeMultipleOfflineStreams(
 SHERPA_ONNX_API typedef struct SherpaOnnxOfflineRecognizerResult {
   const char *text;
 
-  // Pointer to continuous memory which holds timestamps
+  const char *lang;
```
Comment (csukuangfj) [inline sherpa-onnx/c-api/c-api.h]: For the newly added member, please add it as the last field.

That is, move line 499 to line 529

File: sherpa-onnx/c-api/c-api.cc [hunk 2]
```diff
@@ -308,8 +308,27 @@ struct SherpaOnnxOfflineStream {
       : impl(std::move(p)) {}
 };
 
+sherpa_onnx::OfflineRecognizerConfig convertConfig(
```
Comment (csukuangfj) [inline sherpa-onnx/c-api/c-api.cc]:
```suggestion
static sherpa_onnx::OfflineRecognizerConfig convertConfig(
```

File: sherpa-onnx/c-api/c-api.cc [hunk 3]
```diff
@@ -398,17 +417,15 @@ SherpaOnnxOfflineRecognizer *CreateOfflineRecognizer(
...
-
-  SherpaOnnxOfflineRecognizer *recognizer = new SherpaOnnxOfflineRecognizer;
-
-  recognizer->impl =
-      std::make_unique<sherpa_onnx::OfflineRecognizer>(recognizer_config);
+  return recognizer_config;
+}
 
-  return recognizer;
+void SetSherpaOnnxOfflineRecognizerConfig(
```
Comment (csukuangfj) [inline sherpa-onnx/c-api/c-api.cc]:
```suggestion
void SherpaOnnxOfflineRecognizerSetConfig(
```

File: sherpa-onnx/c-api/c-api.h (hunk 1 above)
Comment (iprovalo) [reply sherpa-onnx/c-api/c-api.h]: Done.

File: sherpa-onnx/c-api/c-api.cc [hunk 4]
```diff
@@ -461,6 +478,13 @@ const SherpaOnnxOfflineRecognizerResult *GetOfflineStreamResult(
   pText[text.size()] = 0;
   r->text = pText;
 
+  //lang
+  const auto &lang = result.lang;
+  char *c_lang = new char[lang.size() + 1];
+  std::copy(lang.begin(), lang.end(), c_lang);
+  c_lang[lang.size()] = '\0';
+  r->lang = c_lang;
```
Comment (csukuangfj) [inline sherpa-onnx/c-api/c-api.cc:486]: Remember to use
```
delete[] r->lang;
```
to avoid memory leak.

**Review by csukuangfj — CHANGES_REQUESTED** [review]

**Review by csukuangfj — COMMENTED** [review]
Thank you!

Left some minor comments. Otherwise, it looks good to me.

File: sherpa-onnx/c-api/c-api.cc (hunk 4 above)
Comment (iprovalo) [reply sherpa-onnx/c-api/c-api.cc:486]: Done.

File: sherpa-onnx/c-api/c-api.cc (hunk 2 above)
Comment (iprovalo) [reply sherpa-onnx/c-api/c-api.cc]: Done.

File: sherpa-onnx/c-api/c-api.cc (hunk 3 above)
Comment (iprovalo) [reply sherpa-onnx/c-api/c-api.cc]: Done.

**Comment by iprovalo** [comment]
@csukuangfj I addressed the comments. Thank you for the suggestions!

**Comment by csukuangfj** [comment]
Thank you! We will merge it once the CI gets passed.

(Don't worry about the failed CI tests.)

**Pull request merged**